libcloud/storage/drivers/azure_blobs.py

```diff
@@ -217,15 +221,34 @@ def __init__(self, key, secret=None, secure=True, host=None, port=None,
                                                       port=port, **kwargs)
 
     def _ex_connection_class_kwargs(self):
-        result = {}
-
-        # host argument has precedence
-        if not self._host_argument_set:
-            result['host'] = '%s.%s' % (self.key, AZURE_STORAGE_HOST_SUFFIX)
+        # if the user didn't provide a custom host value, assume we're
+        # targeting the default Azure Storage endpoints
+        if self._host is None:
+            return {'host': '%s.%s' % (self.key, AZURE_STORAGE_HOST_SUFFIX)}
+
+        # connecting to a special storage region like Azure Government or
+        # Azure China requires setting a custom storage endpoint but we
+        # still use the same scheme to identify a specific account as for
+        # the standard storage endpoint
+        try:
+            host_suffix = next(
```

Comment:
Is there a situation where we may want to allow end user to specify full host as-is - aka so we don't add any prefix to it, but user already includes that in the ``host`` argument?

I think this would give users more flexibility in some situations.

Reply:
To clarify - if a new endpoint ever gets added or similar, this would require use to update the code to support it.

If we allowed user to specify a full arbitrary host, they may be able to use that instead (until a new version is released / similar).

For us to be able to support that, we would likely need to add new argument to the constructor - e.g. ``ex_use_host_as_is`` / ``ex_dont_prefix_host`` (or some better name, can't come up with anything better atm).

Reply:
The only current Azure Storage version that I'm aware of which wouldn't be covered by the approach proposed in this PR is [Azure Stack Hub Storage](https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-storage-overview?view=azs-2008). However the API version that libcloud uses (`2018-11-09`) is newer than the [latest supported version](https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-profiles-azure-resource-manager-versions?view=azs-2008#overview-of-the-2019-03-01-hybrid-profile) of Azure Stack (`2017-11-09`) so libcloud doesn't support it anyways.

I get the appeal of an `ex_force_host` or similar argument, but I wonder if the simplicity of this explicit approach will be more user-friendly as it doesn't introduce an argument that's specific to just this driver and instead manages the complexity internally. Thoughts?

Reply:
Yeah, I didn't really mean it as an alternative, but in addition to your proposed changes (so the interface for the common case would still be the same, but in case there is a need to use a fully custom host, there is an easy way to do it).

I also don't think that's a blocker for this PR so if you think it's a good idea, feel free to open a new PR with that change in the future.

Reply:
In that case I'd say to defer the addition of `ex_force_host` for now.